crates/comm/src/peer/outbound.rs

```diff
@@ -62,29 +62,40 @@ pub fn start_peer_sender_worker(
             trace!("receive packet from queue");
             let packet: ProtocolPacket = maybe_break!(app_outbound_rx.recv().await);
 
-            // encode packet
-            trace!("encode packet: {:?}", packet);
-            let buf = try_continue!(try_encode_packet(&packet), "Failed to encode packet");
-
             // these locks may cause some contention - investigate
             let mut packet_number = packet_number.lock().await;
             let mut pending_acks_write = pending_acks.write().await;
 
+            let packets =
+                if let Some(ProtocolPacket::PktRequestAvailablePeers(_)) = packet.packet_type {
+                    vec![SocketPacket::empty(
+                        SocketPacketType::RequestAvailablePeers,
+                        *packet_number,
+                        0,
+                    )]
+                    .into_iter()
+                } else {
+                    // encode packet
+                    trace!("encode packet: {:?}", packet);
+                    let buf = try_continue!(try_encode_packet(&packet), "Failed to encode packet");
+
+                    buf.chunks(MAX_PROTOCOL_PACKET_CHUNK_SIZE).enumerate().map(
+                        |(chunk_idx, chunk)| {
+                            SocketPacket::new(
+                                SocketPacketType::Data,
+                                *packet_number,
+                                chunk_idx as u32,
+                                chunk,
+                            )
+                            .expect("failed to create data packet")
+                        },
+                    );
+                };
             // split packet into network packets and send
-            for net_packet in
-                buf.chunks(MAX_PROTOCOL_PACKET_CHUNK_SIZE)
-                    .enumerate()
-                    .map(|(chunk_idx, chunk)| {
-                        SocketPacket::new(
-                            SocketPacketType::Data,
-                            *packet_number,
-                            chunk_idx as u32,
-                            chunk,
-                        )
-                        .expect("failed to create packet")
-                    })
-            {
+            for net_packet in packets {
```

Comment:
This is a bit strange - we seem to have it implemented at both the socket layer and the protocol layer. It should be one or the other, probably the protocol layer so we are backed by encryption.